### PR TITLE
Remove SAPM exporter, use OTLP exporter

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Remove the SAPM exporter and use OTLP exporter for all trace ingest.

**Testing:**
Regenerated examples. Functional tests already use otlp for trace exports.

**Documentation:**
Replaced mention in README.
Added upgrade guidelines, and a note in NOTES.txt to display. I also added an example showing a situation that will trigger the warning in NOTES.txt.

### CODE REVIEW

**Review by samiura — APPROVED** [review]

File: helm-charts/splunk-otel-collector/templates/NOTES.txt [hunk 1]
```diff
@@ -82,3 +82,7 @@ Splunk OpenTelemetry Collector is installed and configured to send data to Splun
   - Some libraries may be enabled by default. For current status, see: https://github.com/open-telemetry/opentelemetry-operator#controlling-instrumentation-capabilities
   - Splunk provides best-effort support for native OpenTelemetry libraries, and full support for Splunk library distributions. For used libraries, refer to the values.yaml under "operator.instrumentation.spec".
 {{- end }}
+{{- if not (eq (toString ((.Values.agent.config).exporters).sapm) "<nil>") }}
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/NOTES.txt:85]: This should work
```suggestion
{{- if not (eq (toString .Values.agent.config.exporters.sapm) "<nil>") }}
```

**Review by dmitryax — APPROVED** [review]
One comment. Otherwise, LGTM

File: helm-charts/splunk-otel-collector/templates/NOTES.txt (hunk 1 above)
Comment (dmitryax) [reply helm-charts/splunk-otel-collector/templates/NOTES.txt:85]: Also, need to check the same for gateway

Comment (atoulme) [reply helm-charts/splunk-otel-collector/templates/NOTES.txt:85]: No, that fails with `make render`:
```
Error: template: splunk-otel-collector/templates/NOTES.txt:85:32: executing "splunk-otel-collector/templates/NOTES.txt" at <.Values.agent.config.exporters.sapm>: nil pointer evaluating interface {}.sapm
```

**Pull request merged**